openedx/features/qverse_features/registration/signals.py

```diff
@@ -131,17 +129,23 @@ def _create_or_update_qverse_user_profile(edx_user, user_info):
         edx_user (User): An edx user instance whose qverse profile is going to be created/updated
         user_info (dict): User info whose profile is going to be created/updated
     """
-    department = Department.objects.get(number=user_info.get('departmentid'))
+    department = None
+    try:
+        department = Department.objects.get(number=user_info.get('departmentid'))
+    except Department.DoesNotExist:
+        LOGGER.exception('The Department with id {} does not exist.'.format(user_info.get('departmentid')))
+        user_info.update({'status': 'Failed'})
+        raise ValueError('Invalid Department ID')
+
     qverse_profile_data = {
                             'current_level': user_info.get('levelid'),
                             'other_name': user_info.get('othername'),
                             'mobile_number': user_info.get('mobile'),
                             'programme': user_info.get('programmeid'),
                             'department': department,
                             'registration_number': user_info.get('regno')
-                            }
-    _, created = QVerseUserProfile.objects.update_or_create(user=edx_user,
-                                                            defaults=qverse_profile_data)
+                          }
+    _, created = QVerseUserProfile.objects.update_or_create(user=edx_user, defaults=qverse_profile_data)
     if created:
         LOGGER.info('{} qverse profile has been created.'.format(edx_user.username))
         user_info.update({'status': 'Created'})
```

Comment:
no need to update you can simply reassigned value `row['status'] = 'Created'`

Reply:
done.